src/coreclr/src/vm/clsload.cpp

```diff
@@ -3955,7 +3955,7 @@ ClassLoader::LoadTypeHandleForTypeKey_Body(
     }
     EX_HOOK
     {
-        LOG((LF_CLASSLOADER, LL_INFO10, "Caught an exception loading: %x, %0x (Module)\n", pTypeKey->GetTypeToken(), pTypeKey->GetModule()));
+        LOG((LF_CLASSLOADER, LL_INFO10, "Caught an exception loading: %x, %0x (Module)\n", pTypeKey->ComputeHash(), pTypeKey->GetModule()));
```

Comment:
We could still emit the token for regular def types and satisfy the assert. Hashes are awkward to map back to the type unless it shows up elsewhere in the log.
```suggestion
        LOG((LF_CLASSLOADER, LL_INFO10, "Caught an exception loading: %x, %0x (Module)\n", pTypeKey->IsConstructed() ? pTypeKey->ComputeHash() : pTypeKey->GetTypeToken(), pTypeKey->GetModule()));
```

Reply:
Nice synchronization of our responses :-).